docs/usage.md

```diff
@@ -0,0 +1,41 @@
+## Provider configuration
+
+The provider accepts the following configuration attributes under the `provider` block.
```

Comment:
I like the type description, but I think we should still denote whether something is required or optional.  Aligned with https://www.terraform.io/docs/providers/kubernetes/index.html#argument-reference

Reply:
Good point. Everything is optional so I'll just add that as a note before the list. I feel like adding the word "optional" to all the entries just clutters it up? WDYT?

Reply:
Added a line noting everything is optional but a meaningful set of options should be configured.